tests/test_extensions.py

```diff
@@ -172,3 +172,36 @@ def dummy_patch(self, vqip, *args, **kwargs):
 
     assert node.pull_check() == "dummy_patch"
     assert node.pull_set({"volume": 1}) == "dummy_node - 1"
+
+
+def test_custom_class():
+    """Test a custom class."""
+
+    import tempfile
+
+    from wsimod.nodes.nodes import Node, NODES_REGISTRY
+    from wsimod.orchestration.model import Model, to_datetime
+
+    class CustomNode(Node):
+        def __init__(self, name):
+            super().__init__(name)
+            self.custom_attr = 1
+
+        def end_timestep(self):
+            self.custom_attr += 1
+            super().end_timestep()
+
+    NODES_REGISTRY["CustomNode"] = CustomNode
```

Comment:
There's no need to add the new node to the registry. Just subclassing `Node` already does that for you:

https://github.com/ImperialCollegeLondon/wsi/blob/61573a9d2e1997f9d2189a93f3cfbdf9c915638e/wsimod/nodes/nodes.py#L18

```suggestion
```